src/test/ui/traits/operator-overloading-issue-52025.rs

```diff
@@ -0,0 +1,59 @@
+// build-pass
+
+#[cfg(target_arch = "x86")]
+use std::arch::x86::*;
+#[cfg(target_arch = "x86_64")]
+use std::arch::x86_64::*;
```

Comment:
I think this test will fail outside of x86 env, should have `only-x86`.

Reply:
Hmm, I thought that `only-x86` means 32 bit x86 arch, so it would skip x86_64.
Is there a combination of flags to make it check on both x86_32 and x86_64? If not, should we maybe instead check only for x86_64, since it's more popular nowadays?

Reply:
> should we maybe instead check only for x86_64, since it's more popular nowadays?

Sounds good 👍

Reply:
only-x86 I believe covers both i686 (i.e., 32-bit) and x86_64.

Reply:
Oh, TIL! But the issue can be reproducible on both env so I think `only-x86_64` seems enough here.

Reply:
Updated